third_party/ijar/zip_main.cc

```diff
@@ -152,11 +157,16 @@ int extract(char *zipfile, char *exdir, char **files, bool verbose,
     return -1;
   }
 
-  char output_root[PATH_MAX];
+  char output_root[PATH_MAX + 1];
   if (exdir != NULL) {
-    concat_path(output_root, PATH_MAX, cwd.c_str(), exdir);
+    if (!concat_path(output_root, sizeof(output_root), cwd.c_str(), exdir)) {
+      return -1;
+    }
+  } else if (cwd.length() >= sizeof(output_root)) {
+    fprintf(stderr, "current working directory path too long");
+    return -1;
   } else {
-    strncpy(output_root, cwd.c_str(), PATH_MAX);
+    strcpy(output_root, cwd.c_str());
```

Comment:
Linter is not happy with `strcpy`, can you put back `strncpy`?

Reply:
Hum, is this linter why there are so many `strncpy` calls with off-by-one errors in this file? :)